src/engine_core/rendering/src/Vulkan/VulkanAllocatedBuffer.cpp

```diff
@@ -21,7 +24,22 @@ Hush::VulkanVertexBuffer::VulkanVertexBuffer(uint32_t size, VkBufferUsageFlags u
                    "Vertex buffer allocation failed!");
 }
 
-uint32_t Hush::VulkanVertexBuffer::GetSize() const noexcept
+void Hush::VulkanAllocatedBuffer::Dispose(VmaAllocator allocator)
```

Comment:
As for the `dispose` pattern, do you think we should use it or use the destructor instead?

Reply:
The Dispose pattern here I believe fits this use case more, since the disposal code will be ran in a deletion queue, and not necessarily when the object is freed